scripts/get_bracket_data.js

```diff
@@ -0,0 +1,72 @@
+const Contract = require("@truffle/contract")
+const { getOrdersPaginated, printOrder } = require("./utils/to_be_imported_from_dex_contracts.js")
+const { isOnlySafeOwner } = require("./utils/trading_strategy_helpers")
+
+const argv = require("yargs")
+  .option("masterSafe", {
+    type: "string",
+    describe: "Address of Gnosis Safe owning every bracket",
+  })
+  .demand(["masterSafe"])
+  .help(
+    "Make sure that you have an RPC connection to the network in consideration. For network configurations, please see truffle-config.js"
+  )
+  .version(false).argv
+
+module.exports = async callback => {
+  try {
+    // Init params
+    const ERC20Detailed = artifacts.require("ERC20Detailed")
+    const ProxyFactory = artifacts.require("GnosisSafeProxyFactory.sol")
+
+    const proxyFactory = await ProxyFactory.deployed()
+    const eventData = await proxyFactory.getPastEvents("ProxyCreation", { fromBlock: 0, toBlock: "latest" })
+    const safeDeployments = eventData.map(event => event.args[0]).slice(-100)
+    //For easy rinkeby testing use:
+    //bracketTraderAddresses = ['0xc867f926392a4e55b8800afbbbad8c99efd70e49']
+    console.log("In total we have ", safeDeployments.length, " safe deployments")
+    const isDeployedByMaster = await Promise.all(
+      safeDeployments.map(async bracketTrader => await isOnlySafeOwner(argv.masterSafe, bracketTrader, artifacts))
```

Comment:
I believe that `async` and `await` are not needed here.
Interestingly, I tested the time this takes to run with both `.slice(-100)` and `.slice(-200)`, and the time goes up by a factor of about two. However, we are sending the calls in parallel: if you replace this instruction with a for loop it takes about 15 times more. I wonder if we are hitting some request limit on Infura?